R/assignment_linter.R

```diff
@@ -64,63 +72,113 @@
 #'   linters = assignment_linter(allow_pipe_assign = TRUE)
 #' )
 #'
+#' lint(
+#'   text = "x = 1",
+#'   linters = assignment_linter(operator = "=")
+#' )
+#'
 #' @evalRd rd_tags("assignment_linter")
 #' @seealso
 #' - [linters] for a complete list of linters available in lintr.
 #' - <https://style.tidyverse.org/syntax.html#assignment-1>
 #' - <https://style.tidyverse.org/pipes.html#assignment-2>
 #' @export
-assignment_linter <- function(allow_cascading_assign = TRUE,
+assignment_linter <- function(operator = c("<-", "<<-"),
```

Comment:
Positional changes could affect downstream dependencies if the linter is configured using positional arguments. We should at least check for `is.logical(operator)` to stop with a helpful error message.

Reply:
Thanks for flagging. I don't see anyone using positional calls:

https://github.com/search?q=lang%3AR+%2F%5Cbassignment_linter%5B%28%5D%5B%5Ea%29%5D%2F+-path%3A.Rd&type=code

And I think 1st is the correct position for the new argument. Still, it can't hurt to add the safety check you mention.